syntax = "proto3";
option java_package = "com.kyhsgeekcode.minecraftenv.proto";

message ActionSpaceMessageV2 {
  // Discrete actions for movement and other commands as bool
  bool attack = 1;
  bool back = 2;
  bool forward = 3;
  bool jump = 4;
  bool left = 5;
  bool right = 6;
  bool sneak = 7;
  bool sprint = 8;
  bool use = 9;
  bool drop = 10;
  bool inventory = 11;

  // Hotbar selection (1-9) as bool
  bool hotbar_1 = 12;
  bool hotbar_2 = 13;
  bool hotbar_3 = 14;
  bool hotbar_4 = 15;
  bool hotbar_5 = 16;
  bool hotbar_6 = 17;
  bool hotbar_7 = 18;
  bool hotbar_8 = 19;
  bool hotbar_9 = 20;

  // Camera movement (pitch and yaw)
  float camera_pitch = 21;
  float camera_yaw = 22;

  repeated string commands = 23;
}